build/release/build_release.sh

```diff
@@ -142,14 +143,14 @@ popd > /dev/null
 if [ "$SKIP_BUILD" != true ]; then
   export MAVEN_OPTS="-Xmx2048m"
   echo "building release"
-  mvn $MAVEN_FLAGS -DskipTests -Dall clean install
+  mvn $MAVEN_FLAGS -DskipTests -Dall clean generate-sources javadoc:jar install
```

Comment:
I'm wondering if this helps at all, given that release publish does this:

```
mvn clean deploy -DskipTests -Dall
```

See that "clean"? I'm guessing that if you want to deploy javadocs, the building of them should be done there?
Also, the sources are already packaged, don't think we need to bother the build_release script with them, e.g.:

![image](https://github.com/geotools/geotools/assets/325433/61a3ceca-2f45-460b-9979-617ce59231b7)

@jodygarnett knows the release procedures much better than I do, and might provide some more suggestions.

Reply:
> I'm wondering if this helps at all, given that release publish does this:
> 
> ```
> mvn clean deploy -DskipTests -Dall
> ```

It was somewhat hard to figure out how the release process worked. You're absolutely correct that it was unnecessary to add that generate-sources. Not sure why I did that.

Where in the scripts is this mvn-call you are referring to?

Reply:
> 
> It was somewhat hard to figure out how the release process worked. You're absolutely correct that it was unnecessary to add that generate-sources. Not sure why I did that.
> 

I remember now. And it is actually necessary. It isn't the sources.jar that is generated, but the Java-sources we generate from xsd and such. Javadoc-generation will fail without that goal.

Reply:
Can we add comment so we do not have to remember this detail again in the future?